Sources/Reporters/DirectoryTree/DirectoryTreeReporter.swift

```diff
@@ -16,24 +51,55 @@ public struct DirectoryTreesReporter: DiagnosticsReporting {
     let includeHiddenFiles: Bool
     let includeSymbolicLinks: Bool
     let printFullPath: Bool
+    let trunks: [Directory]
 
     /// Creates a new reporter for directory trees.
+    /// eventually uses the new init(trunk: [Directory])
     /// - Parameters:
     ///   - directories: The directories to create trees for.
     ///   - maxDepth: The max depth of directories to visit. Defaults to `.max`.
     ///   - maxLength: The max length of nodes in a directory to handle. Defaults to `10`.
     ///   - includeHiddenFiles: Whether hidden files should be captured. Defaults to `false`.
     ///   - includeSymbolicLinks: Whether symbolic links should be captured. Defaults to `false`.
     ///   - printFullPath: Whether the full path of the node should be printed or just the name. Defaults to `false`.
+    @available(*, deprecated, message: "Use the new init(trunks: [Directory])")
     public init(
         directories: [URL],
         maxDepth: Int = .max,
         maxLength: Int = 10,
         includeHiddenFiles: Bool = false,
         includeSymbolicLinks: Bool = false,
         printFullPath: Bool = false
+    ) {
+        var newTrunks: [Directory] = []
+        directories.forEach { directoryURL in
+            newTrunks.append(Directory(url: directoryURL,
+                                    maxDepth: maxDepth,
+                                    maxLength: maxLength,
+                                    includeHiddenFiles: includeHiddenFiles,
+                                    includeSymbolicLinks: includeSymbolicLinks,
+                                    printFullPath: printFullPath))
+        }
+        self.init(directories: directories, trunks: newTrunks)
+    }
+    
+    /// Creates a new reporter for directory trees
+    /// - Parameter trunks: Array Collection of <Directory> to be diagnosed
+    public init(trunks: [Directory]) {
+        self.init(directories: [], trunks: trunks)
+    }
+    
+    private init(
+        directories: [URL],
+        trunks: [Directory],
+        maxDepth: Int = .max,
+        maxLength: Int = 10,
+        includeHiddenFiles: Bool = false,
+        includeSymbolicLinks: Bool = false,
+        printFullPath: Bool = false
     ) {
         self.directories = directories
```

Comment:
I don't we need `directories` anymore now, so you can remove those from the initialisers.

Reply:
do you mean to remove property "directories"? and if we do that wouldn't that break users existing codebase and cause crash ?

Reply:
The `directories` property is unused and internal, so implementors won't notice the difference. You'll still keep it in the initializer for public access, but since we use your new initializer now we no longer need the stored property 👌

Reply:
"internal" 👌 💯
i totally forgot about the property being internal & mistakenly assumed it as public 🤦‍♂️.

Reply:
PR updated